wp_api/src/lib.rs

```diff
@@ -137,6 +137,54 @@ pub enum JsonValue {
     Object(HashMap<String, JsonValue>),
 }
 
+uniffi::custom_newtype!(WpResponseString, Option<String>);
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(try_from = "BoolOrString")]
+pub struct WpResponseString(pub Option<String>);
+
+// In some cases, WordPress API may return a different type for a field than expected. One example,
+// is when a `false` boolean value is returned when a `String` is expected.
+//
+// We handle these issues by deserializing them into some expected combinations, such as
+// `BoolOrString` and then map them into a new type that wraps the original type. For example,
+// `WpResponseString` is a new type for `Option<String>`, that uses `BoolOrString` to deserialize.
+//
+// During this conversion, there may be some values that are not clear how they should be mapped.
+// For example, when we are expecting a `String` field, if we get a `false` value, we can assume
+// that it's `null`, but there isn't a clear conversion for the `true` value, so we return an
+// error.
+#[derive(Debug, PartialEq, Eq, thiserror::Error, uniffi::Error)]
+pub enum WpApiNewtypeParsingError {
+    #[error("Expecting a `String` value for this field, but received the boolean `true` instead")]
+    BooleanTrueIsReturnedWhenStringIsExpected,
+}
```

Comment:
I'm not sure if a dedicated error type is necessary. For reference, [I used serde's error in the api.wordpress.org implementation](https://github.com/Automattic/wordpress-rs/blob/a452263910aed900276b6e44e2f30f1c433a4e14/wp_api/src/wordpress_org/de.rs#L33-L36).

Reply:
If we are going to throw an error, I like that being one of our own to differentiate from the parsing errors we get from `serde`.

I saw that in your PR, but didn't want to say anything yet, because I figured we'll be working on this a bit more and this error handling might not even remain like this 🤷‍♂️

Reply:
The type would be lost, right? Because serde would convert the error instance to one of the cases in [`serde::de::Error`](https://rust-lang.github.io/hashbrown/serde/de/trait.Error.html), maybe (`custom`)?

I don't mind either way. My approval still stands. Please don't let this comment be the blocker for merging this PR.

Reply:
Yes, when it's used to parse by serde, but the error type is still there when the `TryFrom` is directly used. Again, these are mostly personal preferences to be honest, I don't think there is a right answer 🤷 

In that case, I'll go ahead and merge to keep things moving.